ibis/expr/types/relations.py

```diff
@@ -1693,7 +1693,7 @@ def _assemble_set_op(
             queue.append(node)
         result = queue.popleft()
         assert not queue, "items left in queue"
-        return result.to_expr().select(*self.columns)
+        return result.to_expr()
```

Comment:
Looking at https://github.com/ibis-project/ibis/pull/5882 it looks like the extra projection was added to fix something - do we know what PR(s) fixed the original issue so we can now remove the `select`?

Reply:
I'm not sure yet, wanted to try it out to see whether we have a test case.

Reply:
This was a workaround for a poorly designed set operation IR, where the top level IR (previously select and set ops effectively) wasn't able to be compiled unless the object was ultimately a select.

Now we don't have that restriction and unions/intersections/difference are just another kind of relation.

Reply:
Ah, makes sense, thanks.